pkg/kubectl/cmd/logs.go

```diff
@@ -228,16 +251,37 @@ func (o *LogsOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []str
 	return nil
 }
 
-func (o LogsOptions) Validate() error {
-	logsOptions, ok := o.Options.(*api.PodLogOptions)
+func (o LogsOptions) Validate(args []string) error {
```

Comment:
If you need pass `args` through `LogsOptions` struct. `Validate` should not accept input parameters.